Boa.Constrictor/Screenplay/Conditions/Collection/IsACollectionOfType.cs

```diff
@@ -0,0 +1,95 @@
+namespace Boa.Constrictor.Screenplay
+{
+    /// <summary>
+    /// Methods to check if the collection items satifies the given conditions.
+    ///</summary>
+    ///<example>
+    /// Sample usage:
+    /// Actor.WaitsUntil(TextList.For(ResultPage.ResultLinks), IsAcollectionOfType{string}.AndEveryItem(IsEqualTo.Value(link)));
+    ///</example>
+    public class IsACollectionOfType<T>

```

Comment:
Each of the Conditions that have builder methods in this class use `IEnumerable`, not `ICollection`. I think it would be more precise to name the class `IsAnEnumerable<T>`.